tests/testproject/TestProject.AppHost/TestProgram.cs

```diff
@@ -32,20 +32,32 @@ private TestProgram(string[] args, Assembly assembly, bool includeIntegrationSer
 
         if (includeIntegrationServices)
         {
-            var sqlserver = AppBuilder.AddSqlServerContainer("sqlserver");
-            var mysql = AppBuilder.AddMySqlContainer("mysql");
-            var redis = AppBuilder.AddRedisContainer("redis");
-            var postgres = AppBuilder.AddPostgresContainer("postgres");
-            var rabbitmq = AppBuilder.AddRabbitMQContainer("rabbitmq");
-            var mongodb = AppBuilder.AddMongoDBContainer("mongodb");
+            var sqlserverContainer = AppBuilder.AddSqlServerContainer("sqlservercontainer");
+            var mysqlContainer = AppBuilder.AddMySqlContainer("mysqlcontainer");
+            var redisContainer = AppBuilder.AddRedisContainer("rediscontainer");
+            var postgresContainer = AppBuilder.AddPostgresContainer("postgrescontainer");
+            var rabbitmqContainer = AppBuilder.AddRabbitMQContainer("rabbitmqcontainer");
+            var mongodbContainer = AppBuilder.AddMongoDBContainer("mongodbcontainer");
+            var sqlserverAbstract = AppBuilder.AddSqlServerContainer("sqlserverabstract");
+            var mysqlAbstract = AppBuilder.AddMySqlContainer("mysqlabstract");
+            var redisAbstract = AppBuilder.AddRedisContainer("redisabstract");
+            var postgresAbstract = AppBuilder.AddPostgresContainer("postgresabstract");
+            var rabbitmqAbstract = AppBuilder.AddRabbitMQContainer("rabbitmqabstract");
+            var mongodbAbstract = AppBuilder.AddMongoDB("mongodbabstract");
```

Comment:
@mitchdenny - were these "Abstract" resources all not supposed to have `Container` on them? As this stands, this code spins up 2 SQL Server containers in the exact same way. Same for MySql, Redis, Postgres, RabbitMQ.